BHoMUpgrader41/Converter.cs

```diff
@@ -60,6 +65,100 @@ public Converter() : base()
             return newVersion;
         }
 
+        public static Dictionary<string, object> UpgradeHarpoonSettings(Dictionary<string, object> harpoonSettings)
+        {
+            if (harpoonSettings == null)
+                return null;
+
+            Dictionary<string, object> newHarpoonSettings = new Dictionary<string, object>(harpoonSettings);
+            if (!newHarpoonSettings.ContainsKey("AdvanceConfig"))
+            {
+                newHarpoonSettings.Add("AdvanceConfig", new List<string>());
+            }
+
+            return newHarpoonSettings;
+        }
+
+        public static Dictionary<string, object> UpgradeRadiantSource(Dictionary<string, object> radiantSource)
+        {
+            if (radiantSource == null)
+                return null;
+
+            Dictionary<string, object> energySource = new Dictionary<string, object>(radiantSource);
+            energySource["_t"] = energySource["_t"].ToString().Replace("RadiantSource", "EnergySource");
+
+            if (!energySource.ContainsKey("EnergyFluxInput"))
+            {
+                energySource.Add("EnergyFluxInput", "0 [W m^-2]");
+            }
```

Comment:
Adding properties isn't a problem for versioning so this isn't needed (@adecler can correct me if I'm wrong mind!)

Reply:
In this case, I add a value for the new property. If we don't do it via versioning, will it be a empty string? 
A empty string may cause problem because there is a default value given though normal oM creation and the following functionality does not handle null well.
`public virtual string EnergyFluxInput { get; set; } = "0 [W m^-2]"; `

Reply:
Someone can set the string to be an empty string too, any operation using that object should be handling the case for a user doing that. Therefore there's not real reason to do it in versioning. Especially because later we might change the default of that string, and we would need to remember to change it here too which is double maintenance for no real benefit.

Reply:
That is true. I was handling the check via Create but with upgrade we have another way of "creating" a oM so need to move the check into later stage.
Will remove this tomorrow in case @adecler has some other guidance.